### PR TITLE
Refactor carma-messenger to add web-ui

### PR DESCRIPTION
**Pull request by msmcconnell**
This PR refactors the repo to include the web ui. To clarify the separation of concerns, the top of the repo now has the following structure.

config (Configurations)
doc (Documentation and media)
messenger-core (ROS Nodes)
README.md  
web-ui (Web UI)

In addition the instructions on README.md have been updated to reflect the new setup. 

Supports
 usdot-fhwa-stol/CARMAPlatform#606

### CODE REVIEW

**Comment by maefromm** [comment]
@msmcconnell discussed with @TonysCousin. Having the core and UI in the same repo with 2 docker images works. Right now we expect it to be small/medium repo. However if becomes too large, it may need to be separated out. With this in mind, we should rename "web-ui" to carma-messenger-ui". Eventually in future sprints, the CARMAWebUI will be relabeled to "carmaplatform-web-ui" along with other repos.

**Comment by maefromm** [comment]
@msmcconnell from John's comment "One thing that did grab my attention, is in the readme file, there are several links to specific files in the repo.  Some are pointing to files on the develop branch and at least one is pointing to a file in a feature branch.  For long-term reliability, these should all be pointing to the master branch. Maybe because this repo is so new that's not practical, but during the first formal release those links should all be updated."

**Comment by maefromm** [comment]
@msmcconnell I think we need to discuss the https://github.com/usdot-fhwa-stol/carma-messenger/tree/feature/add_web_ui/messenger-core. I don't understand the content of this folder. I think for Shuwei's purpose, this should be called "truck_inspection_plugin" since that's I think is C++ code. Similar folder structure I assume is like "https://github.com/usdot-fhwa-stol/CARMAPlatform/tree/develop/autoware_plugin"

File: web-ui/website/index.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  License for the specific language governing permissions and limitations under
+  the License.
+-->
+<html>
+<head>
+    <title>CARMA Platform</title>
```
Comment (maefromm) [inline web-ui/website/index.html]: Minor but should say CARMA Messenger

Comment (msmcconnell) [reply web-ui/website/index.html]: Yep

**Comment by msmcconnell** [comment]
Can't seem to respond to each comment from some reason so I'll do them all at once. 
1. Renaming the folders is fine with me. 
2. The README contains links to the carma platform develop branch at the moment for the carma scripts. I don't mind updating those to point to master. The other link is for the development config and is relative so it should update based on whichever branch you are viewing from. I think this makes the most sense as develop would require the most up to date config. 
3. messenger-core is not just the plugin. It is all the backend logic for the messenger system. This includes the message node and j2735_convertor nodes. Calling it "core" seems to be a semi-standard practice with many of the other ros repos we depend on.

**Comment by maefromm** [comment]
@dan-du-car please make sure the docker images build on your VM, if it does then please feel free to approve and merge it. Thanks!

**Comment by msmcconnell** [comment]
@dan-du-car 
These are the instructions
https://github.com/usdot-fhwa-stol/carma-messenger/tree/feature/add_web_ui#example-setup

**Comment by maefromm** [comment]
@dan-du-car please clone the specific feature branch, BEFORE we merge to develop. Let us know if this works or have issues. If any issues then it should be updated on thsi branch.

**Pull request merged**